lib/src/inbox_message_view.dart

```diff
@@ -3,50 +3,21 @@ import 'dart:async';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/services.dart';
 import 'package:flutter/widgets.dart';
-import 'airship.dart';
 
-class InboxMessageView extends StatefulWidget {
-  final InboxMessageViewCreatedCallback onViewCreated;
+class InboxMessageView extends StatelessWidget {
+  final String messageId;
 
   InboxMessageView({
-    Key key,
-    @required this.onViewCreated
+    @required this.messageId,
   });
 
-  @override
-  _InboxMessageViewState createState() => _InboxMessageViewState();
-}
-
-typedef void InboxMessageViewCreatedCallback(InboxMessageViewController controller);
-
-class InboxMessageViewController {
-
-  MethodChannel _channel;
-
-  InboxMessageViewController.init(int id) {
-    _channel =  new MethodChannel('com.airship.flutter/InboxMessageView_$id');
-  }
-
-  Future<void> loadMessage(InboxMessage message) async {
-    if (message == null) {
-      throw ArgumentError.notNull('message');
-    }
-
-    return _channel.invokeMethod('loadMessage', message.messageId);
-  }
-}
-
-class _InboxMessageViewState extends State<InboxMessageView> {
-
   Future<void> onPlatformViewCreated(id) async {
-    if (widget.onViewCreated == null) {
-      return;
-    }
-    widget.onViewCreated(new InboxMessageViewController.init(id));
+    MethodChannel _channel =  new MethodChannel('com.airship.flutter/InboxMessageView_$id');
+    _channel.invokeMethod('loadMessage', messageId);
```

Comment:
where is it handling the message ID changing?

Reply:
the inbox_message_view is a stateless and will build and reload the message when the messageId changes